src/name.rs

```diff
@@ -87,17 +88,13 @@ impl<'a> DNSNameRef<'a> {
     /// Constructs a `DNSNameRef` from the given input if the input is a
     /// syntactically-valid DNS name.
     pub fn try_from_ascii(dns_name: &'a [u8]) -> Result<Self, InvalidDNSNameError> {
-        if !is_valid_reference_dns_id(untrusted::Input::from(dns_name)) {
-            return Err(InvalidDNSNameError);
-        }
-
-        Ok(Self(dns_name))
+        DNSNameRef::try_from(dns_name)
```

Comment:
The problem with the name `try_from` here is that it doesn't indicate what the expected encoding is supposed to be. Should it be UTF-8 or Punycode? In #125, which I plan to update soon, I tried to make it even more explicit by adding "punycode" to the name. I think becoming more explicit about that is better than becoming less explicit, so I don't think the change here is good.

Reply:
Ahh, that makes sense. I removed the commit